pkg/controller/node/node_controller.go

```diff
@@ -995,3 +1022,92 @@ func getErrorString(err error) string {
 	}
 	return ""
 }
+
+// alertIfImportantFilesPending checks to see if there are any important files in the
+// machineconfig that the pool should be moving to, and alerts if there are
+func (ctrl *Controller) alertIfImportantFilesPending(pool *mcfgv1.MachineConfigPool) error {
+
+	// The config we would be going to
+	spec_name := pool.Spec.Configuration.Name
+	// The config we're in right now
+	status_name := pool.Status.Configuration.Name
+
+	// Get the machine config objects
+	statusConfig, err := ctrl.mcLister.Get(status_name)
+	if errors.IsNotFound(err) {
+		glog.V(2).Infof("MachineConfig %v has been deleted", status_name)
+		return err
+	}
+
+	specConfig, err := ctrl.mcLister.Get(spec_name)
+	if errors.IsNotFound(err) {
+		glog.V(2).Infof("MachineConfigPool %v has been deleted", spec_name)
+		return err
+	}
+
+	// Make sure we can coax the objects into ignitionv3
+	statusIgnConfig, err := ctrlcommon.ParseAndConvertConfig(statusConfig.Spec.Config.Raw)
+	if err != nil {
+		return err
+	}
+	specIgnConfig, err := ctrlcommon.ParseAndConvertConfig(specConfig.Spec.Config.Raw)
+	if err != nil {
+		return err
+	}
+
+	// Go through the files and see what is new or different
+	oldFileSet := make(map[string]ign3types.File)
+	for _, f := range statusIgnConfig.Storage.Files {
+		oldFileSet[f.Path] = f
+	}
+	newFileSet := make(map[string]ign3types.File)
+	for _, f := range specIgnConfig.Storage.Files {
+		newFileSet[f.Path] = f
+	}
+	diffFileSet := []string{}
+
+	// First check if any files were removed
+	for path := range oldFileSet {
+		_, ok := newFileSet[path]
+		if !ok {
+			// debug: remove
+			glog.Infof("File diff: %v was deleted", path)
+			diffFileSet = append(diffFileSet, path)
+		}
+	}
+
+	// Now check if any files were added/changed
+	for path, newFile := range newFileSet {
+		oldFile, ok := oldFileSet[path]
+		if !ok {
+			// debug: remove
+			glog.Infof("File diff: %v was added", path)
+			diffFileSet = append(diffFileSet, path)
+		} else if !reflect.DeepEqual(oldFile, newFile) {
+			// debug: remove
+			glog.Infof("File diff: detected change to %v", newFile.Path)
+			diffFileSet = append(diffFileSet, path)
+		}
+	}
+
+	// Go through our list of files and alert if they match our important list
+	for _, path := range diffFileSet {
+		// If this is an important file, trigger its alert
+		for alertFilePath, alertFileDesc := range alertOnPausedFiles {
+			if path == alertFilePath {
```

Comment:
Instead of sending alert for each important file, it will good to send a single alert with important change list we care about.